model/resource/cloud_provider/aws/ec2.yaml

```
@@ -0,0 +1,16 @@
+groups:
+  - id: aws.ec2
+    prefix: aws.ec2
+    type: resource
+    brief: >
+      Resources used by Amazon Elastic Compute Cloud (Amazon EC2).
+    attributes:
+      - id: instance.id
+        type: string
+        brief: >
+          The instance id of an AWS EC2 instance. This is the value
+          provided by the EC2 Metadata endpoint under `instance-id`.
+        note: >
+          When both `host.id` and `aws.ec2.instance.id` are present,
+          they SHOULD be equal.
```

Could this also be phrased as follows to avoid duplication?
```suggestion
          When the value in `host.id` is the same value as the `instance-id`
          returned by the metadata endpoint, only `host.id` should be set.
```

---

Per https://github.com/open-telemetry/semantic-conventions/pull/576#discussion_r1413958977 I believe this is not what we do in similar cases such as with `service.instance.id` and `k8s.pod.name`. I think the situation is very similar here and we should do the same.

---

@arminru PTAL!

---

It certainly makes the logic for both setting and reading the attribute easier. The duplicated string should in transport be taken care of by compression and also backends storing it would be able to deduplicate it as well.

I don't think we ever made a proper policy decision on whether to favor simplicity (for both instrumentation and querying) or deduplication but I believe either is a viable approach to take.

@open-telemetry/specs-semconv-approvers WDYT?

---

> I don't think we ever made a proper policy decision on whether to favor simplicity (for both instrumentation and querying) or deduplication but I believe either is a viable approach to take.

fwiw, we abandoned de-duplication in `network.peer.address` etc (https://github.com/open-telemetry/semantic-conventions/pull/342/commits/328a2c651994548ff73e507fc63349be562ca847) because it made it hard to know whether an attribute is missing because it was a duplicate or because the particular instrumentation doesn't capture it

---

I think this moves in a direction I'm hoping to push resource-detection -

That is a division of responsibilities.  An AWS detector would be responsible for this attribute but NOT host.id.

Instead a Host Detector would generically detect host attributes as best as possible across known host-lookup mechanisms.

I'm writing up thoughts on this now, but I'd be in favor of this PR

---

> I'm writing up thoughts on this now, but I'd be in favor of this PR

@jsuereth Do you have any updates on this?

---

I'm supportive of the addition as originally proposed by @mx-psi